pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java

```diff
@@ -206,8 +210,25 @@ void handleNewTxnResponse(CommandNewTxnResponse response) {
             }
             op.callback.complete(txnID);
         } else {
-            LOG.error("Got new txn for request {} error {}", response.getRequestId(), response.getError());
             handleTransactionFailOp(response.getError(), response.getMessage(), op);
+            if (response.getError() == ServerError.TransactionCoordinatorNotFound) {
```

Comment:
Should retry to send the command after the TC client is connected to the TC? I think it should be better to move to `TransactionMetaStoreHandler.connectionOpened`.